homeassistant/components/template/light.py

```diff
@@ -332,26 +324,39 @@ async def async_turn_on(self, **kwargs):
             self._temperature = kwargs[ATTR_COLOR_TEMP]
             optimistic_set = True
 
-        if ATTR_BRIGHTNESS in kwargs and self._level_script:
-            await self._level_script.async_run(
-                {"brightness": kwargs[ATTR_BRIGHTNESS]}, context=self._context
-            )
-        elif ATTR_COLOR_TEMP in kwargs and self._temperature_script:
-            await self._temperature_script.async_run(
-                {"color_temp": kwargs[ATTR_COLOR_TEMP]}, context=self._context
-            )
-        elif ATTR_WHITE_VALUE in kwargs and self._white_value_script:
-            await self._white_value_script.async_run(
-                {"white_value": kwargs[ATTR_WHITE_VALUE]}, context=self._context
-            )
-        elif ATTR_HS_COLOR in kwargs and self._color_script:
+        on_attrs = {}
+        if self._level_action:
+            on_attrs["brightness_new"] = False
+            on_attrs["brightness"] = self._brightness
+        if self._temperature_action:
+            on_attrs["color_temp_new"] = False
+            on_attrs["color_temp"] = self._color
+        if self._white_value_action:
+            on_attrs["white_value_new"] = False
+            on_attrs["white_value"] = self._white_value
+        if self._color_action:
+            on_attrs["hs_new"] = False
+            on_attrs["hs"] = self._color
+            on_attrs["h"] = int(self._color[0])
+            on_attrs["s"] = int(self._color[1])
+
+        if ATTR_BRIGHTNESS in kwargs and self._level_action:
+            on_attrs["brightness_new"] = True
+            on_attrs["brightness"] = kwargs[ATTR_BRIGHTNESS]
+        if ATTR_COLOR_TEMP in kwargs and self._temperature_action:
+            on_attrs["color_temp_new"] = True
+            on_attrs["color_temp"] = kwargs[ATTR_COLOR_TEMP]
+        elif ATTR_WHITE_VALUE in kwargs and self._white_value_action:
+            on_attrs["white_value_new"] = True
+            on_attrs["white_value"] = kwargs[ATTR_WHITE_VALUE]
+        elif ATTR_HS_COLOR in kwargs and self._color_action:
             hs_value = kwargs[ATTR_HS_COLOR]
-            await self._color_script.async_run(
-                {"hs": hs_value, "h": int(hs_value[0]), "s": int(hs_value[1])},
-                context=self._context,
-            )
-        else:
-            await self._on_script.async_run()
+            on_attrs["hs_new"] = True
+            on_attrs["hs"] = hs_value
+            on_attrs["h"] = int(hs_value[0])
+            on_attrs["s"] = int(hs_value[1])
+
+        await self._on_script.async_run(on_attrs, context=self._context)
```

Comment:
Where all these xx_new key are read?

Reply:
Since all of the keys are always sent, this allows a user to know which things `light.turn_on` was called with if they only want the changes. They are not required but allow more flexibility on the user end—I can be remove them.